# Configuration parameters

# Robot wireframe size parameters
# Lengths representening a wireframe model of the robot. All lengths joint to joint
hip_link_length: 0.055 # Straight line distance of the hip link (horizontal leg link)
upper_leg_link_length: 0.1075 # Straight line distance of the upper leg link, joint to joint
lower_leg_link_length: 0.130 # Straight line distance of the lower leg link, joint to joint
body_width: 0.078 # Horizontal width between hip joints
body_length: 0.186 # Length between shoulder joints

# Stance parameters
default_stand_height: 0.155   # Height of robot body center when standing
stand_front_x_offset: 0.015   # Fwb/back offset of front two legs from default stance
stand_back_x_offset: -0.000      # Fwd/back offset of back two legs from default stance
#stand_front_x_offset: -0.010   # Offset better tuned for trot gait
#stand_back_x_offset: -0.010      # Offset better tuned for trot gait
lie_down_height: 0.083        # Height of body center when sitting
lie_down_foot_x_offset: 0.065 # Fwd/back offset of all(?) feet from default stance when sitting 

# Servo parameters
num_servos: 12
servo_max_angle_deg: 82.5
RF_3: {num: 1,  center: 257,range: 388.7,direction:  1, center_angle_deg:  -18.25}
RF_2: {num: 2,  center: 257,range: 121.0,direction:  1, center_angle_deg: 1.36}
RF_1: {num: 3,  center: 257,range: 245.7,direction: -1, center_angle_deg:  -67.16}
RB_3: {num: 4,  center: 257,range: 388.7,direction:  1, center_angle_deg:  -18.25}
RB_2: {num: 5,  center: 257,range: 139.3,direction:  1, center_angle_deg: 1.18}
RB_1: {num: 6,  center: 257,range: 234.7,direction:  1, center_angle_deg:  79.45}
LB_3: {num: 7,  center: 257,range: 66,direction:  1, center_angle_deg: 2.50}
LB_2: {num: 8,  center: 257,range: 469.3,direction:  1, center_angle_deg:  0.35}
LB_1: {num: 9,  center: 257,range: 432.7,direction: -1, center_angle_deg:  -45.38}
LF_3: {num: 10, center: 257,range: 275.0,direction:  1, center_angle_deg: 4.20}
LF_2: {num: 11, center: 257,range: 304.3,direction:  1, center_angle_deg: 93.80}
LF_1: {num: 12, center: 257,range: 275.0,direction:  1, center_angle_deg:  1.20}

# Control Parameters
transit_tau: 0.3 # Time constant in seconds for first order filters during transition state
transit_rl: 0.06 # Body motion rate limit in m/s for transition state
transit_angle_rl: 0.35 # Body motion angular rate limit in rad/s for transition state 
max_fwd_velocity: 0.4
max_side_velocity: 0.4
max_yaw_rate: 0.35

# Gait parameters
z_clearance: 0.050 # Max height of foot during swing phase. Increase to have feet go
                   # higher during swing. Feet travel in a triangular motion during swing
alpha: 0.5 # Controls whether feet are centered within fwd/back motion
beta: 0.5 # Controls whether feet are centered within side to side motion
foot_height_time_constant: 0.02

##################
# GAIT SELECTION #
##################
# Uncomment one set of gait parameters for the desired
# gait to use, and comment out the other set. 

############################################
############## 8 Phase Gait ################
############################################
# Only one leg is ever in swing phase at one time
# 4 swing phases + 4 body motion phases to balance body over
# three legs that will remain in contact with ground
num_phases: 8
rb_contact_phases: [1, 0, 1, 1, 1, 1, 1, 1]
rf_contact_phases: [1, 1, 1, 0, 1, 1, 1, 1]
lf_contact_phases: [1, 1, 1, 1, 1, 1, 1, 0]
lb_contact_phases: [1, 1, 1, 1, 1, 0, 1, 1]
overlap_time: 0.0
swing_time: 0.36
body_shift_phases: [1, 2, 3, 4, 5, 6, 7, 8]
fwd_body_balance_shift: 0.035 # Amount the body center will shift forward when walking
back_body_balance_shift: 0.005 # Amount the body center will shift backward when walking
side_body_balance_shift: 0.015 # Amount the body center will shift side to side when walking 

#################################################
############## 4 Phase Trot Gait ################
#################################################
# 4 Phase gait where the two diagonally oposite legs move in
# tandem together during swing phase
# 2 swing phases + 2 phases where all four feet are on the ground
# num_phases: 4
# rb_contact_phases: [1, 0, 1, 1]
# rf_contact_phases: [1, 1, 1, 0]
# lf_contact_phases: [1, 0, 1, 1]
# lb_contact_phases: [1, 1, 1, 0]
# overlap_time: 0.18
# swing_time: 0.18
# body_shift_phases: [0,0,0,0]
# fwd_body_balance_shift: 0
# back_body_balance_shift: 0
# side_body_balance_shift: 0

# TF Parameter
lidar_x_pos: 0.045 # X offset of lidar coordinate frame center from robot body center
lidar_y_pos: 0.0 # Y offset of lidar coordinate frame center from robot body center
lidar_z_pos: 0.085 # Z offset of lidar coordinate frame center from robot body center
lidar_yaw_angle: 180 # Angle in degrees of yaw angle of the mounted lidar. Pitch and roll are assumed to be 0

# Node parameters
dt: 0.02 # Loop time of main loop, seconds (0.02 s is 50 hz) 
publish_odom: true  # Determines whether odometry is published or not

# Debug mode
debug_mode: false # Prints out additional information
plot_mode: false # Publishes data for the python plotting node to work

# Standalone mode
standalone_mode: false # Enables node to run without i2cpwnboard

